index.js

```diff
@@ -0,0 +1,414 @@
+ * @throws DataError - If the currency is not supported by PayPal.
+ * @throws DataError - If the amount's value is not a number.
+ *
+ * @returns {object} A formatted amount.
+ */
+const formatAmount = ({amount}) => {
+  const supported = currencies.supported.has(amount.currency_code);
+  if(!supported) {
+    throw new BedrockError(
+      `Unsupported PayPal currency ${amount.currency_code}`,
+      Errors.Data, {public: true}
+    );
+  }
+  const bigAmount = BigNumber(amount.value);
+  const notNumber = bigAmount.toString() === 'NaN';
+  if(notNumber) {
+    throw new BedrockError(
+      `Invalid amount ${amount.value} ${amount.currency_code}`,
+      Errors.Data, {public: true}
+    );
+  }
+  if(currencies.noDecimal.has(amount.currency_code)) {
+    amount.value = bigAmount.toFixed(0).toString();
+    return amount;
+  }
+  amount.value = bigAmount.toFixed(2).toString();
+  return amount;
+};
+
+/**
+ * Gets a PayPal auth token so we can call on the api.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} [options.clientId = paypal_client_id] - PayPal Client Id.
+ * @param {string} [options.secret = paypal_secret] - PayPal Secret.
+ *
+ * @returns {Promise<object>} The data returned with an `access_token`.
+ */
+const getAuthToken = async ({clientId, secret, api}) => {
+  const authUrl = `${api}/v1/oauth2/token`;
+  const options = {
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'Accept-Language': 'en_US'
+    },
+    auth: {
+      username: clientId,
+      password: secret
+    }
+  };
+  const body = 'grant_type=client_credentials';
+  const {data} = await axios.post(authUrl, body, options);
+  return data;
+};
+
+/**
+ * Gets the options needed to make a secure paypal request.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} [options.clientId = paypal_client_id] - PayPal Client Id.
+ * @param {string} [options.secret = paypal_secret] - PayPal Secret.
+ *
+ * @returns {Promise<object>} Settings for paypal.
+ */
+const getOptions = async ({clientId, secret, api} = getConfig()) => {
+  const {access_token, token_type} = await getAuthToken(
+    {clientId, secret, api});
+  const Authorization = `${token_type} ${access_token}`;
+  const headers = {
+    Authorization,
+    Accept: 'application/json'
+  };
+  return {api, headers};
+};
+
+/**
+ * Gets an order from PayPal to ensure the order was made.
+ *
+ * @param {object} options - Options to use.
+ * @param {object} options.order - The order from the client.
+ *
+ * @returns {Promise<object|null>} The order data from PayPal.
+ */
+const getOrder = async ({id}) => {
+  try {
+    const {headers, api} = await getOptions();
+    const options = {headers};
+    const url = `${api}/v2/checkout/orders/${id}`;
+    const {data} = await axios.get(url, options);
+    return data;
+  } catch(e) {
+    logger.error('getOrder not found', {error: e});
+    if(e.response.status === 404) {
```

Comment:
one of the reasons I took to using apisauce module is because of axios error wonkiness: https://github.com/axios/axios#handling-errors

Unfortunately, you can't count on `e.response` existing and therefore you can hit an error attempting to access `status` of `undefined`.

https://github.com/infinitered/apisauce

Reply:
I looked into using API Sauce. It looks great, is backwards compatible with axios, etc. however I could not easily find a way of doing old school auth with it. So I'm going to skip on this one unless time allows.